assets/js/stripe.js

```diff
@@ -879,43 +807,50 @@ jQuery( function( $ ) {
 		},
 
 		/**
-		 * Queries the server for a PaymentIntent.
-		 *
-		 * @param {string} savedCard The ID of the saved card. (Optional)
-		 * @return {Promise} A promise that resolves to a payment intent.
+		 * Handles changes in the hash in order to show a modal for PaymentIntent confirmations.
 		 */
-		getIntent: function( savedCard ) {
-			savedCard = savedCard || 'new';
+		onHashChange: function() {
+			var partials = window.location.hash.match( /^#?confirm-pi-(.+):(.+)$/ );
 
-			return new Promise( function( resolve, reject ) {
-				// If we have an intent for this card already, use it.
-				if ( wc_stripe_form.paymentIntentCache.hasOwnProperty( savedCard ) ) {
-					return resolve( wc_stripe_form.paymentIntentCache[ savedCard ] );
-				}
+			if ( ! partials || 3 > partials.length ) {
+				return;
+			}
 
-				$.ajax( {
-					url: wc_stripe_form.getAjaxURL( 'create_intent' ),
-					type: 'post',
-					data: {
-						saved_card: ( 'new' === savedCard ? null : savedCard ),
-					},
-					dataType: 'json',
-					success: function( result ) {
-						if ( result.error ) {
-							return reject( result.error );
-						}
-
-						wc_stripe_form.paymentIntentCache[ savedCard ] = result;
-						resolve( result );
-					},
-					error: function() {
-						reject( {
-							code   : 'ajax_failed',
-							message: wc_stripe_params.payment_intent_error,
-						} );
-					},
+			var intentClientSecret = partials[1];
+			var successRedirectURL = decodeURIComponent( partials[2] );
+
+			wc_stripe_form.openIntentModal( intentClientSecret, successRedirectURL );
+
+			// Cleanup the URL
```

Comment:
Should this move up above at least the openItentModal just to ensure the hash is cleaned up even if the wc_stripe_form.openIntentModal call explodes?